test/goose/console_test.clj

```diff
@@ -0,0 +1,69 @@
+(ns goose.console-test
+  (:require [clojure.test :refer [deftest is testing use-fixtures]]
+            [goose.broker :as broker]
+            [goose.brokers.redis.console :as redis-console]
+            [goose.client :as c]
+            [goose.console :as console]
+            [goose.test-utils :as tu]
+            [goose.worker :as w]
+            [hickory.core :as h]
+            [hickory.select :as h-select]
+            [ring.mock.request :as mock])
+  (:import (java.lang AbstractMethodError IllegalArgumentException)))
+
+(use-fixtures :each tu/redis-fixture tu/reset-called?-fixture)
+
+(deftest dispatch-handler-test
+  (testing "Should dispatch to redis handler when app-handler is given redis broker details"
+    (with-redefs [redis-console/handler (tu/stub redis-console/handler)]
+      (is (= (tu/get-called?) false))
+      (console/app-handler {:broker       tu/redis-producer
+                            :app-name     ""
+                            :route-prefix "/goose/console"}
+                           (mock/request :get "/goose/console/"))
+      (is (= (tu/get-called?) true)))))
+
+(deftest homepage-view-test
+  (let [worker (w/start tu/redis-worker-opts)
+        job-opts (assoc tu/redis-client-opts :retry-opts (assoc tu/retry-opts :max-retries 0))
```

Comment:
Move these inside the deftest.

Reply:
I am assuming, you are suggesting to move the worker and job-opts from the deftest block to the testing block, I have moved them.